### DISCUSSION TITLE
关于更新1.6版本后的占用内存

### DISCUSSION THREAD

**Discussion by LeafRains** [libnyanpasu/clash-nyanpasu#1500] [Q&A] [answered]
各位大佬好，我今天更新了1.6版本后，发现主界面采用了Web View 2框架，在笔记本上打开主界面时内存占用**高达1.5G**，想问下这是合理的吗。

![image](https://github.com/user-attachments/assets/4a435da8-1bcf-4c8e-b1f6-89517a31cd9a)

如果这是正常情况的话，有什么办法可以减少一些内存占用呢？我打开了减轻动画选项，但似乎内存占用也无法减少太多。似乎和我代理组较多有关？我如果切换到仅有10个代理组的配置文件，占用就约400M；而切换到39个代理组的配置文件，占用就会到1.5G。

先感谢各位大佬的解答。

**Comment by greenhat616** [comment]
请提供一份 profile 以便调查

**Reply by LeafRains** [reply]
好的，附件中是我的profile，由于github本身不支持上传yaml文件，故而我将其后缀改为了TXT。

[ProfilesForUL.txt](https://github.com/user-attachments/files/16902193/ProfilesForUL.txt)


节点信息是用的我个人的proxies providers，所以文件中隐去了其更新的url。节点数量为220个。

![image](https://github.com/user-attachments/assets/0bdb6a9f-9a7a-47f3-94c9-7d7b8548a0cf)

**Reply by greenhat616** [reply]
方便配置邮箱发 [i@elaina.moe](mailto:i@elaina.moe)么。目前无法复现 1.5G 的内存占用情况。200-500M 占用是正常的，主要是 MUI 的组件问题。

**Reply by LeafRains** [reply]
请问还需要提供什么文件呢，是需要包含节点信息的配置文件吗。不好意思，最近比较忙，一直没看回复。
包含规则配置、节点信息和软件设置文件的邮件已发，具体版本1.6.0和1.6.1均有此问题。

**Comment by zhangbobo131415** [comment]
<img width="942" alt="image" src="https://github.com/user-attachments/assets/8d86bd7d-5728-4675-ace9-c2636f450434">
我也有同样的内存占用问题，版本是1.6.1 至于如何复现，好像用一会就这样了

**Reply by zhangbobo131415** [reply]
@greenhat616 配置已发邮箱

**Comment by zhangbobo131415** [comment]
<img width="870" alt="image" src="https://github.com/user-attachments/assets/f10fe8ad-487a-45c0-9c11-40df277b98ae">
不断变大！

**Reply by greenhat616** [reply]
可以先试一下开发版吗？看看能不能有所缓解

**Reply by zhangbobo131415** [reply]
@greenhat616 
这个是和内核有关吗？我使用另一个软件基本j可以稳定在200M以下
<img width="803" alt="image" src="https://github.com/user-attachments/assets/487d573a-15bc-4000-9137-8906d142413e">

**Reply by greenhat616** [reply, marked as answer]
否，和 react-dom-router 的内存泄露问题有关，开发版中有进行针对性重构，但是还在测试，你可以先测试一下，看看能不能缓解这个问题。此外，mui 的 DOM 组件过度嵌套，以及基于 JS 的弹性动画都需要额外的内存开销，所以理想中 200M 内存是需要的。
https://github.com/libnyanpasu/clash-nyanpasu/releases/tag/pre-release

**Reply by zhangbobo131415** [reply]
好的，thanks

**Reply by LeafRains** [reply]
应该和内核没关系，因为如果不打开界面的话内存占用是正常的。实际上看Windows进程的话，内核进程和界面进程是分开的，前者占用是比较稳定的，Mac上应该也差不多。

**Reply by greenhat616** [reply]
> 应该和内核没关系，因为如果不打开界面的话内存占用是正常的。实际上看Windows进程的话，内核进程和界面进程是分开的，前者占用是比较稳定的，Mac上应该也差不多。

最近的开发版有尝试么？profile 已经发送的话，等他有空会继续跟进。

**Reply by zhangbobo131415** [reply]
<img width="951" alt="image" src="https://github.com/user-attachments/assets/4c5d7499-5b34-4e5d-9349-7f040a7ff895">
目前来看[Clash.Nyanpasu_2.0.0-alpha+6903480_aarch64.dmg](https://github.com/libnyanpasu/clash-nyanpasu/releases/download/pre-release/Clash.Nyanpasu_2.0.0-alpha+6903480_aarch64.dmg) 这个版本貌似没有内存泄漏了，用了一个多小时，内存都在200左右

**Reply by LeafRains** [reply]
> > 应该和内核没关系，因为如果不打开界面的话内存占用是正常的。实际上看Windows进程的话，内核进程和界面进程是分开的，前者占用是比较稳定的，Mac上应该也差不多。
> 
> 最近的开发版有尝试么？profile 已经发送的话，等他有空会继续跟进。

尝试了一下开发版，确实有所改善，界面内存基本稳定在1G左右，期待后面能够继续优化。感谢各位开发者大大！！
![image](https://github.com/user-attachments/assets/2579e2d2-98ed-4d7c-8165-0aa7bac9031b)

更新：发现当界面位于代理组，并静置超过五分钟后，内存占用还是会突然上去，只能期待各位开发者大大的继续优化更新了。
![image](https://github.com/user-attachments/assets/41cf67b6-a980-4573-8713-1dfa0e777c25)

**Reply by greenhat616** [reply]
本地尝试你之前发的 profile，进行测试的时候有一次会触发高内存占用，其他时候开了 performance 测试就无法再次复现了。因此新版中的泄露原因，目前还无法下结论。此外，我们已经注意到了 JS Heap 会比少 groups 多出 200M 的占用。目前还没时间去分析这个。

但这个应该是一个实现上的内存泄露问题，或是上游的 bug。可以考虑开个 issue 方便后续追踪这个问题。

**Reply by LeafRains** [reply]
好的，已开[issue](https://github.com/libnyanpasu/clash-nyanpasu/issues/1703)

**Comment by LeafRains** [comment]
这个我知道，其实占用内存异常的是界面，不打开就是正常的。

zhangbobo131415 ***@***.***> 于2024年9月27日周五 15:16写道：

**Comment by keiko233** [comment]
在我们这边由于使用大量的js计算动画以及mui组件库使得前端开销过大，看上去动画设计与高性能无法兼得。

于是我们正在准备开发新的material you design组件库来彻底解决这一问题，在我们人手缺乏的情况下，组件库的开发是需要花费较长周期，详细的进展后续会逐渐公布

**Reply by LeafRains** [reply]
感谢各位开发者大大的辛苦劳动
